syntax = "proto3";

package banking;


message PrepareCreateAccountRequest {
  string transaction_id = 1;
  string name = 2;
  string email = 3;
  double initial_balance = 4;
}

message PrepareCreateAccountResponse {
  bool vote_commit = 1;
  string reason = 2;        // optional: explanation if vote_commit == false
}

message CommitCreateAccountRequest {
  string transaction_id = 1;
}

message AbortCreateAccountRequest {
  string transaction_id = 1;
}

message Empty {
}

// Node-level 2PC participant API
service TwoPCParticipant {
  rpc PrepareCreateAccount(PrepareCreateAccountRequest)
      returns (PrepareCreateAccountResponse);

  rpc CommitCreateAccount(CommitCreateAccountRequest)
      returns (Empty);

  rpc AbortCreateAccount(AbortCreateAccountRequest)
      returns (Empty);
}
